dace/codegen/tools/get_hip_arch.cpp

```diff
@@ -17,9 +16,16 @@ int main(int argc, char **argv) {
     if (hipGetDeviceProperties(&prop, i) != hipSuccess ||
         (prop.major == 99 && prop.minor == 99))
       continue;
-    std::stringstream ss;
-    ss << prop.major * 10 << prop.minor;
-    architectures.insert(ss.str());
+
+    // Find architecture, ignoring anything that succeeds a colon
+    void *colon = memchr(prop.gcnArchName, ':', 256);
+    if (colon)
+      *(char *)colon = '\0';
+
+    // Trim starting "gfx"
+    std::string str = prop.gcnArchName + 3;
```

Comment:
These things are inevitable, but maybe check if the string starts with "gfx"?

Reply:
You’re right. I’ll keep the gfx here and remove its addition in the cmake file

Reply:
Fixed